packages/filesystem/src/browser/file-tree/file-tree-model.ts

```diff
@@ -148,8 +149,13 @@ export class FileTreeModel extends TreeModelImpl implements LocationService {
     }
 
     async copy(source: URI, target: Readonly<FileStatNode>): Promise<URI> {
-        const targetUri = target.uri.resolve(source.path.base);
+        let targetUri = target.uri.resolve(source.path.base);
         try {
+            if (source.path.toString() === target.uri.path.toString()) {
+            const parent = await this.fileService.resolve(source.parent);
```

Comment:
something is wrong with the formatting

Reply:
can you explain?

Reply:
there is no indentation for the if block. VS Code should do formatting on save.

Reply:
the `if` indentation fixed.
formatting happens in the same way in duplicate, see this: https://github.com/eclipse-theia/theia/blob/57b4b671f621dc1c5c6eec7f8c6ab1fe0585e4a5/packages/workspace/src/browser/workspace-duplicate-handler.ts#L67